src/back/hlsl/writer.rs

```diff
@@ -1388,6 +1388,7 @@ fn builtin_str(built_in: crate::BuiltIn) -> &'static str {
         // fragment
         Bi::FragDepth => "SV_Depth",
         Bi::FrontFacing => "SV_IsFrontFace",
+        // Bi::PrimitiveId => "SV_PrimitiveID",
```

Comment:
why is this commented out?

Reply:
I could certainly uncomment it, but I haven't tested the HLSL path at all.  I will do that.  Speaking of backends, I don't have an Apple device or any knowledge of Metal, so I won't be able to work on that platform.  I can certainly throw together a Linux VM to work on the OpenGL backend at some point.

Reply:
HLSL is emerging now, and we have all the things checked on CI which we care not to break. So don't afraid to be bold on HLSL and MSL. If CI is happy, we are likely happy.
Speaking of which, we'll need to add the use of primitive ID to one of the test. I think "extra.wgsl" should be this, given that it's extra functionality.